### ISSUE TITLE
Fix site deployment workflow to the ASF

### ISSUE THREAD

**Issue by jcamachor** [apache/incubator-xtable#375]
Recently @kywe665 did some changes to have a first version of the site that is compliant with ASF (#352).
However, the preexisting workflow to publish the website, which can be found [here](https://github.com/apache/incubator-xtable/blob/main/.github/workflows/site-deploy.yml), is not working anymore.

Following the [ASF proposed guidelines](https://infra.apache.org/project-site.html), I have created a repository at https://github.com/apache/incubator-xtable-site . I have set it up and pushed the contents generated for the current site in `gh-pages` branch manually.

We should fix the workflow though. I hope that with minor changes to the following block in the `site-deploy` action, we might be able to push the changes to the new repo:
```yaml
      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        if: github.ref == 'refs/heads/main'
        with:
          deploy_key: ${{ secrets.DEPLOY_KEY }}
          publish_dir: ./website/build
```

**Comment by jcamachor** [comment]
@zabetak , @jbonofre , @vinothchandar , any insights based on your engagement in other ASF projects? This is an area I have not checked for some time and things have changed a bit.

@zabetak , I was checking the [Calcite deployment file](https://github.com/apache/calcite/blob/main/.github/workflows/publish-non-release-website-updates.yml), do you recall where CALCITE_WEBSITE_BUILD secret is coming from?

**Comment by zabetak** [comment]
@jcamachor The `CALCITE_WEBSITE_BUILD` token was requested in https://issues.apache.org/jira/browse/INFRA-23044 we have to file a similar request if we want to push changes to `incubator-xtable-site` repo.

**Comment by jcamachor** [comment]
Fixed in #398.

**Comment by jcamachor** [comment]
Related: I have opened a ticket to disable the 'pages-build-deployment' GH action.
https://issues.apache.org/jira/browse/INFRA-25652
